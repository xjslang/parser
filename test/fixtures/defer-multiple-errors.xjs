function testMultipleFailures() {
  console.log('start')
  
  defer {
    console.log('defer 1 executes')
  }
  
  defer {
    console.log('defer 2 - will fail')
    throw new Error('second error')
  }
  
  defer {
    console.log('defer 3 - will also fail')
    throw new Error('third error')
  }
  
  defer {
    console.log('defer 4 executes')
  }
  
  console.log('end')
}

testMultipleFailures()
