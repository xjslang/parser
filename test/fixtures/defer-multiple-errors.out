start
end
defer 4 executes
defer 3 - will also fail
Error: third error
defer 2 - will fail
Error: second error
defer 1 executes
